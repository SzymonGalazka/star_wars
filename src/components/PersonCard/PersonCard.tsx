import React from "react";
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import { PersonDataType } from '../../utils/playerModels';

const useStyles = makeStyles({
  root: {
    maxWidth: 200,
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
  winner: {
    color: '#2E8B57',
  },
});

const PersonCard = (props: PersonDataType & { isWinner: boolean } ) => {
  const classes = useStyles();

  return (
    <div style={{ display: "inline-block" }} data-testid="person_card">
      <Card className={classes.root}>
        <CardContent>
          <Typography className={classes.title} color="textSecondary" data-testid="person_card_name">
            <b>Name: </b> {props.name}
          </Typography>
          <Typography className={classes.pos} color="textSecondary" data-testid="person_card_gender">
            <b>Gender: </b> {props.gender}
          </Typography>
          <Typography className={classes.pos} color="textSecondary" data-testid="person_card_mass">
            <b>Mass: </b> {props.mass}
          </Typography>
          {props.isWinner && <Typography className={classes.winner} color="textSecondary" data-testid="person_card_winner">
            <b> Wins! </b>
          </Typography>}
        </CardContent>
      </Card>
    </div>
  );
}

export default PersonCard;