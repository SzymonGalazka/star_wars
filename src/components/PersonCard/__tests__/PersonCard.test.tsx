import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render, getByTestId } from "@testing-library/react";

import PersonCard from '../PersonCard';

describe("PersonCard", () => {
  it("should render PersonCard component", () => {
    const tree = TestRenderer.create(<PersonCard name="testPersonName" gender="male" mass={80} isWinner={true} />);
    expect(tree).toMatchSnapshot();
  });

  it("should have name displayed", () => {
    const { container } = render(
      <PersonCard name="testPersonName" gender="male" mass={80} isWinner={true} />
    );
    const element = getByTestId(container, 'person_card_name')
    expect(element).toHaveTextContent(
      "testPersonName"
    );
  });

  it("should have model displayed",() => {
    const { container } = render(
      <PersonCard name="testPersonName" gender="male" mass={80} isWinner={true} />
    );
    const element = getByTestId(container, 'person_card_gender')
    expect(element).toHaveTextContent(
      "male"
    );
  });

  it("should have crew number displayed", () => {
    const { container } = render(
      <PersonCard name="testPersonName" gender="male" mass={80} isWinner={true} />
    );
    const element = getByTestId(container, 'person_card_mass')

    expect(element).toHaveTextContent(
      "80"
    );
  });

  it("should display winner section", () => {
    const { container } = render(
      <PersonCard name="testPersonName" gender="male" mass={80} isWinner={true} />
    );
    const element = getByTestId(container, 'person_card_winner')
    expect(element).toHaveTextContent(
      "Wins!"
    );

  });
  
});