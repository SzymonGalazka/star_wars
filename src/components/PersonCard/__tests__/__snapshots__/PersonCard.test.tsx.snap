// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonCard should render PersonCard component 1`] = `
<div
  data-testid="person_card"
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <div
    className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-title-2 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="person_card_name"
      >
        <b>
          Name: 
        </b>
         
        testPersonName
      </p>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="person_card_gender"
      >
        <b>
          Gender: 
        </b>
         
        male
      </p>
      <p
        className="MuiTypography-root makeStyles-pos-3 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="person_card_mass"
      >
        <b>
          Mass: 
        </b>
         
        80
      </p>
      <p
        className="MuiTypography-root makeStyles-winner-4 MuiTypography-body1 MuiTypography-colorTextSecondary"
        data-testid="person_card_winner"
      >
        <b>
           Wins! 
        </b>
      </p>
    </div>
  </div>
</div>
`;
